src/TransferList/TransferList.types.ts

```diff
@@ -1,14 +1,79 @@
-export type TransferListProps = {
+export type TransferListItem =
+  | string
+  | { primaryLabel: string; secondaryLabel?: string; id: string };
+
+export type TransferListProps<T> = {
+  /**
+   * The key to be used for the filter.
+   */
+  filterKey?: (item: T) => string;
+  /**
+   * Control the transfer of the items
+   */
+  handleTransfer?: (value: string[], intent: "toTarget" | "toSource") => void;
   /**
    * Array of Items.
    */
-  items?: string[];
+  items: TransferListItem[] | T[];
+  /**
+   * Array of keys for the items on the right side.
+   */
+  targetListKeys?: string[];
+  /**
+   * Source list label
+   */
+  sourceListLabel: string;
+  /**
+   * Target list label
+   */
+  targetListLabel: string;
+  /**
+   * Props for the default item
+   */
+  itemProps?: {
+    /**
+     * The key to be used for the primary label
+     */
+    primaryLabel: (item: T) => string;
+    /**
+     * The key to be used for the secondary label
+     */
+    secondaryLabel?: (item: T) => string;
+  };
+  /**
+   * Callback fired when the items are transferred.
+   */
+  onTransfer?: (value: string[], intent: "toTarget" | "toSource") => void;
+};
+
+export type SingleListProps = {
+  /**
+   * Checked items
+   */
+  checked: string[];
+  /**
+   * Item array
+   */
+  items: {
+    /**
+     * Unique identifier
+     */
+    key: string;
```

Comment:
Here you call it `key` but in ItemProps you call it `id`. Can we be consistent?

Reply:
Converted all to key